src/analysis/inner_product_space/basic.lean

```diff
@@ -427,6 +431,15 @@ lemma real_inner_smul_right {x y : F} {r : ℝ} : ⟪x, r • y⟫_ℝ = r * ⟪
 lemma inner_smul_real_right {x y : E} {r : ℝ} : ⟪x, (r : 𝕜) • y⟫ = r • ⟪x, y⟫ :=
 by { rw [inner_smul_right, algebra.smul_def], refl }
 
+lemma inner_smul {x : E} {r : 𝕜} : @inner 𝕜 E _ (r • x) = r† • @inner 𝕜 E _ x :=
+funext (λ _, inner_smul_left)
+
+lemma real_inner_smul {x : F} {r : ℝ} : @inner ℝ F _ (r • x) = r • @inner ℝ F _ x :=
+funext (λ _, inner_smul_left)
```

Comment:
What's the point in having this lemma? It's just a special case of the one above.

Reply:
It makes rewriting easier, because you don't have to `simp` away the identity (there are versions for the `_left` and `_right` lemmas, too. I've added `real_inner_smul` and `inner_smul_real` just to be consistent with the existing API.

Reply:
Oh, I missed the `†`!